### ISSUE TITLE
Deadwood IPv6/NoIPv6 building fix

### ISSUE THREAD

**Issue by ktrace** [samboy/MaraDNS#97]
configure script seek $DEADWOOD_DIR/src/Makefile.ipv6 but it absent, but Makefile.noip6 present.
```
diff --git a/configure b/configure
index 9e98ac8..fedbd87 100755
--- a/configure
+++ b/configure
@@ -174,14 +174,14 @@ if [ -z "$IPV6" ] ; then
 	cp tcp/Makefile.authonly tcp/Makefile
 	cp dns/Makefile.authonly dns/Makefile
 	cp tools/Makefile.authonly tools/Makefile
+	DEADWOOD_DIR=$( echo deadwood-* )
+	cp $DEADWOOD_DIR/src/Makefile.noipv6 $DEADWOOD_DIR/src/Makefile
 else
 	cp server/Makefile.ipv6 server/Makefile
 	cp qual/Makefile.nothreads qual/Makefile
 	cp tcp/Makefile.ipv6 tcp/Makefile
 	cp dns/Makefile.authonly dns/Makefile
 	cp tools/Makefile.ipv6 tools/Makefile
-	DEADWOOD_DIR=$( echo deadwood-* )
-	cp $DEADWOOD_DIR/src/Makefile.ipv6 $DEADWOOD_DIR/src/Makefile
 fi
```

**Comment by samboy** [comment]
Thanks for the detailed and clued bug report. 

I think, here in 2021, we should change things so that the default is to have IPv6 for everything except the mingw port (The mingw environment I use to make Windows binaries *does* have IPv6, but is missing key stuff like `inet_pton()`, so we would need to add a bunch of extra code to give Deadwood ipv6 there). 

My plan is to ignore the `--ipv6` option given to configure, add `--noipv6`, and if neither is specified, add a “WARNING: ipv6 support not specified; adding IPv6” message.

(There is also some messiness about how the mingw port uses the appropriate Makefile to compile Deadwood I should clean up; the Makefile to use should be set up in the configure script, but is done elsewhere)

**Comment by ktrace** [comment]
As a gentoo maintainer of MaraDNS I will be happy if MaraDNS understand --ipv6 (or --noipv6, no difference) flag to correct obey global gentoo flags. 
BTW, take a look on current patches: https://github.com/gentoo/gentoo/blob/0a8b522380b9d27c119b91be05a5be995730be7e/net-dns/maradns/files/maradns-3.5.0020-flags-ipv6.patch
I may create issues if they got chance to fix.

**Comment by samboy** [comment]
Fixed; MaraDNS now understands both `--ipv6` and `--noipv6` parameters, but defaults to `--ipv6` (with warning) if neither specified (except for the mingw32 build, which still doesn’t have IPv6 support [1] except for coLunacyDNS).

[1] Depending on what you mean by “IPv6” support. Even the mingw32 build supports bogus IPv6 replies and blocklisting IPv6 answers. But the version of mingw32 I use doesn’t have `ip_pton()`, so I would need to put my own version of `ip_pton()` in to Deadwood before it can properly do stuff with IPv6. And, oh, glueness NS referrals are messy w.r.t. IPv6 upstream DNS servers and Deadwood always assumes a glueless NS referral is for a server with an IPv4 record.

**Comment by samboy** [comment]
@ktrace Is this an issue important enough for me to do a MaraDNS 3.5.0021 release over. If I don’t hear back, I’ll wait for some other changes (or a few months, if nothing else happens) before making MaraDNS 3.5.0021; if it will help on your side, I can make a release in the next two weeks or so (probably sooner).

**Comment by samboy** [comment]
Resolved: I will wait before making MaraDNS 3.5.0021

**Comment by samboy** [comment]
MaraDNS 3.5.0021 released.
